deploy-tool/main.go

```diff
@@ -235,27 +289,22 @@ func createManifest(
 		if err != nil {
 			return fmt.Errorf("error when creating batch signing public key: %v", err)
 		}
-		newBatchSigningPublicKeys[name] = *batchSigningPublicKey
+		manifestWrapper.SpecificManifest.BatchSigningPublicKeys[name] = *batchSigningPublicKey
 	}
 
-	manifestWrapper.SpecificManifest.BatchSigningPublicKeys = newBatchSigningPublicKeys
-
-	certificatesByNamespace := manifest.PacketEncryptionKeyCSRs{}
-	newCertificates := manifest.PacketEncryptionKeyCSRs{}
-	for name, packetEncryptionCertificate := range manifestWrapper.SpecificManifest.PacketEncryptionKeyCSRs {
-		if packetEncryptionCertificate.CertificateSigningRequest != "" {
-			newCertificates[name] = packetEncryptionCertificate
-			continue
+	for name, packetEncryptionCertificateSigningRequest := range manifestWrapper.SpecificManifest.PacketEncryptionKeyCSRs {
+		if packetEncryptionCertificateSigningRequest.CertificateSigningRequest != "" {
+			// We never create CSRs in Terraform, so this should never happen
+			return fmt.Errorf("unexpected packet encryption key CSR in Terraform output")
```

Comment:
might be worthwhile to indicate `name` in this Errorf since it is already formatted.
```suggestion
			return fmt.Errorf("unexpected packet encryption key CSR in Terraform output for name %s", name)
```